spec/webauthn/authenticator_attestation_response_spec.rb

```diff
@@ -39,7 +39,7 @@
     it "returns the credential" do
       credential = attestation_response.credential
 
-      expect(credential.id.class).to eq(String)
+      expect(credential.id.class).to eq(BinData::String)
```

Comment:
Is a bit strange to see an expectation of a bindata string.

What we actually care is that it behaves like a string or something like that. I guess what's important is that whatever is returned here is something that can be stored in a string column in a database by the end user...

Reply:
Maybe we can use something like `is_a` or `kind_of` assertion...?

Reply:
It does feel weird to test against `BinData::String`. However, `credential.id` neither `is_a?(String)` nor `kind_of?(String)`.
According to the `bindata` documentation, to get a primitive Ruby object we should use `#snapshot`. This would solve the problem but I wonder whether to redefine the `id` method to use `#snapshot` (which implies renaming the already existing `id` to something not so straightforward like `bindata_id` or something like that) or to just call `id.snapshot` when creating the credential on `AttestedCredentialData#credential`

Reply:
I also wonder if this could be a problem with the rest of the bindata-defined readers

Reply:
On second thought I don't think we should be to concerned about the specific class of the `id`. I think the original test implementation is to blame and this should actually check the expected value, not the class.

Reply:
We'll probably need to let the `FakeClient` receive a fixed credential, to override the randomly generated, so that we can check the expected values then?

Reply:
This is probably out of the scope of this PR.
What about leaving the assertion against `BinData::String` and just adding a `# TODO` comment mentioning this?